### PR TITLE
Vary inflight limit by fidelity level

### PR DESCRIPTION
**Pull request by dasormeter**
Instead of static value of 1 for all latency tests and value of 40 for throughput PerformanceEnvelope tests, vary the inflight limit by the different fidelity levels, slightly larger than the throughput number

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PerformanceEnvelope.scala [hunk 1]
```diff
@@ -216,7 +237,6 @@ trait PerformanceEnvelope {
               ledgerId = participant.ledgerId,
               begin = Some(offset),
               end = None,
-              verbose = false,
```
Comment (fabiotudone-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PerformanceEnvelope.scala:219]: Why this change?

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PerformanceEnvelope.scala [hunk 2]
```diff
@@ -333,15 +355,13 @@ object PerformanceEnvelope {
...
-    * @param numPings  how many pings to run during the throughput test
-    * @param maxInflight how many inflight commands we can have. set it high enough such that the system saturates, keep it low enough to not hit timeouts.
+    * @param numPings       how many pings to run during the throughput test
     * @param numWarmupPings how many pings to run before the perf test to warm up the system
     */
   class ThroughputTest(
       val logger: Logger,
       val envelope: Envelope,
       val numPings: Int = 200,
-      val maxInflight: Int = 40,
```
Comment (fabiotudone-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PerformanceEnvelope.scala:344]: Was this unused?

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PerformanceEnvelope.scala [hunk 3]
```diff
@@ -21,55 +21,73 @@ import com.daml.ledger.api.v1.command_completion_service.{
...
 import scala.util.{Failure, Random, Success, Try}
-import com.daml.ledger.test.performance.{PingPong => PingPongModule}
-import org.slf4j.Logger
 
 sealed trait Envelope {
   val name: String
   val transactionSizeKb: Int
   val throughput: Int
   val latencyMs: Int
+  val inFlightLimit: Int
```
Comment (meiersi-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PerformanceEnvelope.scala:45]: @dasormeter : AFAIS, in a successful test for a specific envelope it  always holds that `numInflightRequests <= throughput * latencyMs / 1000`. So, I'd suggest to not introduce the inFlightLimit property, but instead add a function `maxRequestsInflight` that computes it as per the above formula.

Comment (meiersi-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PerformanceEnvelope.scala:45]: i.e.  `maxRequestsInFlight = throughput * latencyMs / 1000`

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PerformanceEnvelope.scala (hunk 2 above)
Comment (meiersi-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PerformanceEnvelope.scala:344]: @dasormeter : also as we are at it: we probably should compute the number of pings as well from the Envolope. There the goal I'd see is that we get a stable measurement of the throughput. Which means we'd like to observe at least some `X` number of requests sent while maximum throughput is reached.

To see why a constant does not suffice, think about a throughput and latency Envolope that requires more than 200 requests to be in flight at the same time; e.g., the Enterprise one.

So, a better value is to set  `numPings = 10 * maxRequestsInFlight`, which means we send enough pings to observe at least 10 flights of requests. For statistical significance 10 is probably rather on the low side. It matches though quite closely the 200 pings assumption that @daravep took in his original code.

If we see a lot of variation in our measurements, then we should upgrade the 10.

Please also add a comment in the code capturing the reasoning in this comment, if/when you change the code.

**Comment by SamirTalwar-DA** [comment]
@dasormeter, is this still a thing?

**Comment by dasormeter** [comment]
> @dasormeter, is this still a thing?

yeah this still needs work, but has been on the back burner -- moved it back to draft
